SELearning.API/Controllers/CommentController.cs

```diff
@@ -11,48 +12,128 @@ namespace SELearning.API.Controllers;
 public class CommentController : ControllerBase
 {
     private readonly ILogger<CommentController> _logger;
-    private readonly ICommentRepository _repository;
+    private readonly ICommentService _service;
 
-    public CommentController(ILogger<CommentController> logger, ICommentRepository repository)
+    public CommentController(ILogger<CommentController> logger, ICommentService service)
     {
         _logger = logger;
-        _repository = repository;
+        _service = service;
     }
 
     [Authorize]
     [HttpGet("{ID}")]
-    [ProducesResponseType(typeof(CommentDTO), 200)] // OK
+    [ProducesResponseType(typeof(Comment), 200)] // OK
     [ProducesResponseType(404)] // Not Found
-    public async Task<ActionResult<CommentDTO>> GetComment(int id)
-        => (await _repository.GetAsync(id)).ToActionResult();
+    public async Task<ActionResult<Comment>> GetComment(int id)
+    {
+        try
+        {
+            return Ok(await _service.GetCommentFromCommentId(id));
+        }
+        catch (Exception)
```

Comment:
What happens here in case of an internal server error?

Reply:
@JoachimBorup Det er din kode

Reply:
The only way for me to know whether the `IContentService` fails or not is catching a generic `Exception` thrown by the service.
If an internal server error is thrown as an exception, then it's unfortunately caught and "presented" with the Not Found error code, and we'll probably need to make some custom exceptions as discussed [here](https://github.com/BDSA-2021/BDSA21-Say-AAAAAA/pull/28#discussion_r762416765), or figure something else out. If it's presented as something else than an exception, we can talk about it at the next meeting.